generator/gpt2/gpt2_generator.py

```diff
@@ -60,16 +61,24 @@ def prompt_replace(self, prompt):
         return prompt
 
     def result_replace(self, result, actions):
-        # print("\n\nBEFORE RESULT_REPLACE:")
-        # print(repr(result))
-        
+        debug_print = False
+        if debug_print:
+            print("\n\nBEFORE RESULT_REPLACE WITH cut_actions=",self.cut_actions,":")
+            print(repr(result))
+
         result = result.replace('."', '".')
         result = result.replace("#", "")
         result = result.replace("*", "")
         result = result.replace("\n\n", "\n")
-        result = re.sub("(?<=\w)\.\.(?:\s|$)", ".", result)
+        result = re.sub(r"(?<=\w)\.\.(?:\s|$)", ".", result)
         # result = first_to_second_person(result)
-        result = cut_trailing_sentence(result)
+        result = cut_trailing_sentence(result, cut_actions=self.cut_actions)
+        if not self.cut_actions:
+            lines = [l.rstrip() for l in result.rstrip().split("\n")]
+            lines = [l for l in lines if len(l) > 0]
+            lines = [l.replace(">", "").strip() if ('"' in l or "?" in l or "you say" in l.lower() or "you ask" in l.lower()) and i < len(lines)-1 else l for i,l in enumerate(lines)] # Turn dialogue actions into regular lines (if it's not the last one) to keep conversation intact
+            lines = [l for l in lines if not l.startswith(">")] # Remove any remaining actions (but keep results)
+            result = "\n".join(lines)

```

Comment:
key logic for replacement; now when cut_trailing_sentence does not trim actions, we filter non-dialogue actions out while preserving everything else.